internal/app/singularity/config_global_linux.go

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package singularity
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/sylabs/singularity/internal/pkg/buildcfg"
+	"github.com/sylabs/singularity/pkg/runtime/engine/config"
+	"golang.org/x/sys/unix"
+)
+
+// GlobalConfigOp defines a type for a global configuration operation.
+type GlobalConfigOp uint8
+
+const (
+	// GlobalConfigSet is the operation to set a configuration directive value.
+	GlobalConfigSet GlobalConfigOp = iota
+	// GlobalConfigUnset is the operation to unset a configuration directive value.
+	GlobalConfigUnset
+	// GlobalConfigGet is the operation to get a configuration directive value.
+	GlobalConfigGet
+	// GlobalConfigReset is the operation to reset a configuration directive value.
+	GlobalConfigReset
+)
+
+func contains(slice []string, val string) bool {
+	for _, s := range slice {
+		if s == val {
+			return true
+		}
+	}
+	return false
+}
+
+func generateConfig(file string, directives config.Directives, dry bool) error {
+	out := os.Stdout
+
+	configTmpl := filepath.Join(buildcfg.SOURCEDIR, "etc/conf/singularity.conf")
```

Comment:
`buildcfg.SOURCEDIR` seems like something that could be unset or set to a nonexistent location if singularity is not installed from source